File: javascript/gigasecond/gigasecond.js
class Gigasecond {
  constructor(date) {
    this.startDate = new Date(date)
  }

  date() {
    return new Date(this.startDate.getTime() + Gigasecond.GIGA_MILLISECONDS)
  }
}

Gigasecond.GIGA_MILLISECONDS = Math.pow(10, 9) * 1000

module.exports = Gigasecond

File: go/gigasecond/gigasecond.go
package gigasecond

import (
	"math"
	"time"
)

const testVersion = 4

var gigaSecond float64 = math.Pow(10, 9)

func AddGigasecond(t time.Time) time.Time {
	d := time.Duration(gigaSecond) * time.Second
	return t.Add(d)
}
